generator/generator_tests/relation_tags_tests.cpp

```diff
@@ -0,0 +1,263 @@
+    {"ref", "F-16"}},
+  OsmElement::EntityType::Way);
+
+  // Process road tags using relation tags.
+  RelationTagsWay rtw;
+
+  rtw.Reset(10, &road10);
+  rtw(1, reader);
+
+  rtw.Reset(11, &road11);
+  rtw(1, reader);
+
+  // Verify ways tags.
+  TEST_EQUAL(road10.GetTag("ref"), "E-99", ());
+  TEST_EQUAL(road11.GetTag("ref"), "F-16", ());
+}
+
+UNIT_TEST(Process_route_with_ref_network) {
+  /* Prepare relations data:
+   * Relation 1:
+   *   - type = route
+   *   - route = road
+   *   - ref = SP60
+   *   - network = IT:RA
+   *   - members: [
+   *       Way 10:
+   *         - highway = motorway
+   *         - name = Via Corleto
+   *       Way 11:
+   *         - highway = motorway
+   *         - ref = SP60
+   *     ]
+   */
+
+  // Create relation.
+  std::vector<RelationElement::Member> testMembers = {{10, ""},
+                                                      {11, ""}};
+
+  RelationElement e1;
+  e1.m_ways = testMembers;
+  e1.m_tags.emplace("type", "route");
+  e1.m_tags.emplace("route", "road");
+  e1.m_tags.emplace("ref", "SP60");
+  e1.m_tags.emplace("network", "IT:RA");
+
+  std::unordered_map<generator::cache::Key, RelationElement> m_IdToRelation = {{1, e1}};
+  TestOSMElementCacheReader reader(m_IdToRelation);
+
+  // Create buildings polygons.
+  auto road10 = MakeOsmElement(10, {{"highway", "motorway"},
+    {"name", "Via Corleto"}},
+  OsmElement::EntityType::Way);
+
+  auto road11 = MakeOsmElement(11, {{"highway", "motorway"},
+    {"ref", "SP62"}},
+  OsmElement::EntityType::Way);
+
+  // Process road tags using relation tags.
+  RelationTagsWay rtw;
+
+  rtw.Reset(10, &road10);
+  rtw(1, reader);
+
+  rtw.Reset(11, &road11);
+  rtw(1, reader);
+
+  // Verify ways tags.
+  TEST_EQUAL(road10.GetTag("ref"), "IT:RA/SP60", ());
+  TEST_EQUAL(road11.GetTag("ref"), "SP62", ());
+}
+
+UNIT_TEST(Process_associatedStreet) {
+  /* Prepare relations data:
+   * Relation 1:
+   *   - type = associatedStreet
+   *   - name = Main street
+   *   - wikipedia = en:Main Street
+   *   - place =
+   *   - members: [
+   *       Way 2:
+   *         - building = yes
+   *         - addr:housenumber = 121
+   *       Way 3:
+   *         - building = house
+   *         - addr:housenumber = 123
+   *         - addr:street = The Main Street
+   *     ]
+   */
+
+  // Create relation.
+  std::vector<RelationElement::Member> testMembers = {{2, "house"},
+                                                      {3, "house"}};
+
+  RelationElement e1;
+  e1.m_ways = testMembers;
+  e1.m_tags.emplace("type", "associatedStreet");
+  e1.m_tags.emplace("name", "Main Street");
+  e1.m_tags.emplace("wikipedia", "en:Main Street");
+
+  std::unordered_map<generator::cache::Key, RelationElement> m_IdToRelation = {{1, e1}};
+  TestOSMElementCacheReader reader(m_IdToRelation);
+
+  // Create buildings polygons.
+  auto buildingWay2 = MakeOsmElement(2, {{"building", "yes"},
+    {"addr:housenumber", "121"}},
```

Comment:
One line. Or, if its too long, prefer the following syntax:
```
{
{},
{},
};
```
It allows inserting new values without modifying the first and last lines.

Reply:
Reformatter to single line.